plugins/reboot/reboot.go

```diff
@@ -50,24 +102,51 @@ func (plugin) NewTaskPlugin(options plugins.TaskPluginOptions) (plugins.TaskPlug
 		return nil, engines.ErrContractViolation
 	}
 
+	pl.runningTasks.RLock()
 	return taskPlugin{
 		TaskPluginBase: plugins.TaskPluginBase{},
-		reboot:         p.Reboot,
+		plugin:         pl,
+		rebootWhenDone: p.RebootWhenDone,
 	}, nil
 }
 
 func (tp taskPlugin) Dispose() error {
```

Comment:
We need to change the reciever type to `func (tp *taskPlugin) ...` in all the methods... As we no want to pass by reference...
I guess we should stop using the other pattern, just because of the footgun it introduces later...

Note: before `tp` was strictly read-only.